src/java.base/share/classes/jdk/internal/foreign/abi/Binding.java

```diff
@@ -362,13 +370,28 @@ static Binding.Builder builder() {
     class Builder {
         private final List<Binding> bindings = new ArrayList<>();
 
+        private static boolean isSubIntType(Class<?> type) {
+            return type == boolean.class || type == byte.class || type == short.class || type == char.class;
+        }
+
         public Binding.Builder vmStore(VMStorage storage, Class<?> type) {
+            if (isSubIntType(type)) {
```

Comment:
So, this is interesting, in the sense that a cast binding is never emitted in isolation, but always in combination with a vm load /store binding. Does it mean that maybe creating a separate binding for this is conceptually wrong, and maybe it's the semantics of vmstore/load that should be updated? It's more of a theoretical question, in the sense that the code is obviously doing the correct thing as is.

Reply:
I think it makes sense to have cast as a separate binding. Because we don't also do a cast, I think it's nice that it is somehow reflected in the recipe.

I mostly but the code here for the sake of convenience. The alternative would be to add this kind code in each CallArranger implementation, which seemed like unnecessary repetition.

Since this is a builder class, it made sense to me to put it here (as in the class encapsulates some construction logic). Though, maybe now that these methods do more than just a VMStore and VMLoad, they should be renamed as well to something that reflects that. (maybe `erasedVMStore` / `erasedVMLoad`).